src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double RESIZE_COUNT = 1.5;
+    private Object[] elementData;
+    private int defaultSize;
+    private int count;
+
+    public ArrayList() {
+        defaultSize = 10;
+        elementData = new Object[defaultSize];
+        count = 0;
+    }
+
+    private boolean rangeCheckForCount(int index) {
+        if (index >= count || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound of array.");
+        }
+        return true;
+    }
+
+    private void checkArrayLength(int count) {
+        if (count >= defaultSize) {
+            Object[] localeData = elementData;
+            elementData = new Object[defaultSize *= RESIZE_COUNT];
+            System.arraycopy(localeData, 0, elementData, 0, localeData.length);
+        }
+    }
+
+    public boolean checkValue(T element, int i) {
+        if (element != null && element.equals(elementData[i])) {
+            return true;
+        }
```

Comment:
Unnecessary return true statement. You can make this code shorter. If I were you I will rewrite this method as: if (element == null) {return element == elementData[i]} else { return element.equals(elementData[i])}.